tables/cost_optimization_recommendation/cost_optimization_recommendation_table.go

```diff
@@ -0,0 +1,84 @@
+package cost_optimization_recommendation
+
+import (
+	"time"
+
+	"github.com/rs/xid"
+	typehelpers "github.com/turbot/go-kit/types"
+	"github.com/turbot/pipe-fittings/v2/utils"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source_config"
+	"github.com/turbot/tailpipe-plugin-sdk/constants"
+	"github.com/turbot/tailpipe-plugin-sdk/row_source"
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+	"github.com/turbot/tailpipe-plugin-sdk/table"
+)
+
+const CostOptimizationRecommendationsTableIdentifier = "aws_cost_optimization_recommendation"
+
+// CostOptimizationRecommendationsTable - table for CostOptimizationRecommendations
+type CostOptimizationRecommendationsTable struct{}
+
+// Identifier implements table.Table
+func (t *CostOptimizationRecommendationsTable) Identifier() string {
+	return CostOptimizationRecommendationsTableIdentifier
+}
+
+// GetSourceMetadata implements table.Table
+func (t *CostOptimizationRecommendationsTable) GetSourceMetadata() []*table.SourceMetadata[*CostOptimizationRecommendation] {
+	defaultArtifactConfig := &artifact_source_config.ArtifactSourceConfigImpl{
+		FileLayout: utils.ToStringPointer("%{DATA:prefix}/%{DATA:exportName}/%{DATA:folderName}/%{DATA:timestamp}/%{DATA}.csv.gz"),
+	}
+
+	return []*table.SourceMetadata[*CostOptimizationRecommendation]{
+		{
+			// any artifact source
+			SourceName: constants.ArtifactSourceIdentifier,
+			Options: []row_source.RowSourceOption{
+				artifact_source.WithDefaultArtifactSourceConfig(defaultArtifactConfig),
+				artifact_source.WithArtifactExtractor(NewCostOptimizationRecommendationExtractor()),
+			},
+		},
+		{
+			// any artifact source
+			SourceName: constants.ArtifactSourceIdentifier,
+			Options: []row_source.RowSourceOption{
+				artifact_source.WithArtifactExtractor(NewCostOptimizationRecommendationExtractor()),
+			},
+		},
+	}
+}
+
+// EnrichRow implements table.Table
+func (t *CostOptimizationRecommendationsTable) EnrichRow(row *CostOptimizationRecommendation, sourceEnrichmentFields schema.SourceEnrichment) (*CostOptimizationRecommendation, error) {
+	// initialize the enrichment fields to any fields provided by the source
+	row.CommonFields = sourceEnrichmentFields.CommonFields
+
+	// Record standardization
+	row.TpID = xid.New().String()
+	row.TpIngestTimestamp = time.Now()
+
+	if row.LastRefreshTimestamp != nil {
```

Comment:
If we always get this field, we shouldn't have an `if`

Reply:
Done!